### ISSUE TITLE
Crash and freeze at login page

### ISSUE THREAD

**Issue by azmankudus** [avafinger/nanopi-a64-firmware#5]
Hi, I follow new image instructions and flashed RC2 into my card. Then I cannot login after boot. It crash and freeze on login page. My keyboard entry and mouse movement also not detected. Also not able to open tty.

**Comment by avafinger** [comment]
Please, in order to help you i need more information:

a) Can you flash a new SD CARD? Which brand and size is your current SD CARD and the new one?
b) Can you boot without keyboard and mouse and attach it after the login screen appears? Wait a bit longer and see if the blue led is blinking (heartbeat)
* attach first mouse and see if you can move the pointer
* attach second the keyboard, and see if you can type one letter

c) Can you access a Terminal from the Login screen? Press Ctrl+Alt+F1 (or F2 or 3 and so on F7 is the where the GUI is running) in case keyboard is wirking.

And if everything above does not work, can you attach a TTL serial debug and capture the boot log info?
Please refer to the USB TTL debug as described here: https://www.cnx-software.com/2016/07/20/getting-started-with-nanopi-neo-development-board/

**Comment by avafinger** [comment]
Additional info:
I think I was able to reproduce the problem, i usually use a USB 2.0 hub to connect mouse and keyboard and then connect the USB hub to the lower USB port and this always work. I then removed the USB hub and connected mouse and keyboard and i was **NOT** able to use them. I re-connected the USB hub and was still not working. A patch was applied for the **ehci**, that must be the problem, i will check that.
I can see the connection and right after a  disconnection when plugged in the mouse for example:
```
[    7.215784] ehci_irq: highspeed device connect
[   76.247444] ehci_irq: highspeed device disconnect
[   93.139398] ehci_irq: highspeed device connect
[   93.244094] ehci_irq: highspeed device disconnect

```

Do you have a USB hub to connect to **lower** USB port and check if mouse and keyboard works?
Anyway, i will **update** kernel and add **ssh** to be able to connect to the board, the board actually did **not hang or freeze**, just the keyboard and mouse have been forced to disconnect, you can check that watching the clock time at upper right corner for a few minutes.

**Comment by azmankudus** [comment]
hi, i connect keyboard and mouse using usb hub and now working. But i'm not going to use usb hub since i have a few units and not willing to buy more hub without utilising the existing hubs. 

And can you create core version without any DE or share some instruction to build it.

**Comment by avafinger** [comment]
Yes, i can build ubuntu-minimal and you add things you need. But this will not fix the USB issue.
If you wish to build a kernel you can use this kernel: https://github.com/longsleep/linux-pine64 , only missing rtl8189es , but you have a chance to test the mouse and keyboard alone and report back if it works.

**Comment by avafinger** [comment]
I have finally fixed the USB issues. It is now possible to use both USB connectors with and without USB hub. I will push new firmware and you can test it.

**Comment by avafinger** [comment]
Please, flahs a new image with the fix and check if works for you: https://github.com/avafinger/nanopi-a64-firmware#instructions-for-new-image

See Build v2.
